tests/api/runtimes/base.py

```diff
@@ -0,0 +1,207 @@
+from copy import deepcopy
+from mlrun.config import config as mlconf
+import json
+from mlrun.model import new_task
+
+logger = create_logger(level="debug", name="test-runtime")
+
+
+class TestRuntimeBase:
+    def setup_method(self, method):
+        self.namespace = mlconf.namespace = "test-namespace"
+        self._logger = logger
+        self.project = "test-project"
+        self.name = "test-function"
+        self.run_uid = "test_run_uid"
+        self.image_name = "mlrun/mlrun:latest"
+        self.artifact_path = "/tmp"
+
+        self._logger.info(
+            f"Setting up test {self.__class__.__name__}::{method.__name__}"
+        )
+
+        self.custom_setup()
+
+        self._logger.info(
+            f"Finished setting up test {self.__class__.__name__}::{method.__name__}"
+        )
+
+    def custom_setup(self):
+        pass
+
+    def _generate_task(self):
+        return new_task(
+            name=self.name, project=self.project, artifact_path=self.artifact_path
+        )
+
+    def _mock_create_namespaced_pod(self):
+        def _generate_pod(namespace, pod):
+            terminated_container_state = client.V1ContainerStateTerminated(
+                finished_at=datetime.now(timezone.utc), exit_code=0
+            )
+            container_state = client.V1ContainerState(
+                terminated=terminated_container_state
+            )
+            container_status = client.V1ContainerStatus(
+                state=container_state,
+                image=self.image_name,
+                image_id="must-provide-image-id",
+                name="must-provide-name",
+                ready=True,
+                restart_count=0,
+            )
+            status = client.V1PodStatus(
+                phase=PodPhases.succeeded, container_statuses=[container_status]
+            )
+            response_pod = deepcopy(pod)
+            response_pod.status = status
+            response_pod.metadata.name = "test-pod"
+            response_pod.metadata.namespace = namespace
+            return response_pod
+
+        get_k8s().v1api.create_namespaced_pod = unittest.mock.Mock(
+            side_effect=_generate_pod
+        )
+
+    def _execute_run(self, runtime, **kwargs):
+        # Reset the mock, so that when checking is create_pod was called, no leftovers are there (in case running
+        # multiple runs in the same test)
+        get_k8s().v1api.create_namespaced_pod.reset_mock()
+
+        runtime.run(
+            name=self.name,
+            project=self.project,
+            artifact_path=self.artifact_path,
+            **kwargs,
+        )
+
+    def _assert_labels_exist_in_pod_creation(self, labels: dict):
+        expected_labels = {
+            "mlrun/class": "job",
+            "mlrun/name": self.name,
+            "mlrun/project": self.project,
+            "mlrun/tag": "latest",
+        }
+
+        for key in expected_labels:
+            assert labels[key] == expected_labels[key]
```

Comment:
I would simply call it `_assert_labels` 
We will expect the `mlrun/class` label to be set to `job` only for the kubejob, so I would move the definition of the "expected" to the `TestKubejobRuntime`
Also use deepdiff, no need to do dict assertion manually

Reply:
I don't want to do deepdiff since there (potentially) may be other labels created which I'm not asking about, so I don't want to start excluding them. So, I just go over the expected labels and ensure all of them exist there.
I will implement the other changes.